client/ide/lib/firebaserealtimemanager.coffee

```diff
@@ -0,0 +1,373 @@
+kd               = require 'kd'
+KDObject         = kd.Object
+IDEMetrics       = require './idemetrics'
+generatePassword = require 'app/util/generatePassword'
+GOOGLE_OATH_SCOPES = 'email, https://www.googleapis.com/auth/drive.metadata.readonly'
+RealtimeManagerFactory = require './realtimemanagerfactory'
+
+module.exports = class FirebaseRealtimeManager extends RealtimeManagerFactory
+
+  constructor: (@KodingApp) ->
+    super options, data
+    @firebase = new Firebase("https://KodingApp.firebaseio.com")
```

Comment:
`firebase` references in this file are not defined too.

firebase.database namespace doesn't have any named `Event`. https://firebase.google.com/docs/reference/js/

Method names are still derived from Google Drive API.